README.md

```diff
@@ -8,16 +8,207 @@ Local.
+        - [Conflict Control](#conflict-control)
+    - [Model](#model)
+        - [Map between model and document](#map-between-model-and-document)
+        - [Model-specific configuration](#model-specific-configuration)
+        - [_rev property](#_rev-property)
+- [Setup Cloudant Instance](#setup-cloudant-instance)
+- [Installation](#installation)
+- [Configuration](#configuration)
+    - [Generate Datasource](#generate-datasource)
+    - [Datasource Config(`url` vs `username` & `password`)](#datasource-configurl-vs-username--password)
+    - [Requests Plugin](#requests-plugin)
+    - [Example Usage](#example-usage)
+- [CRUD](#crud)
+    - [Update](#update)
+- [Migration](#migration)
+    - [autoupdate vs automigrate](#autoupdate-vs-automigrate)
+    - [isActual](#isactual)
+    - [Example Code](#example-code)
+- [Discovery](#discovery)
+- [Query](#query)
+- [Testing](#testing)
+- [More Info](#more-info)
+- [Feature backlog](#feature-backlog)
+    - [View](#view)
+    - [Index](#index)
+
+<!-- /TOC -->
+
+# Getting Started
+
+## Design
+Loopback tries best to fit its model to a specific database's design, while limited by the nature of database, it's not always possible to support all Loopback features perfectly, and user would like to be aware of some key features about Cloudant before they start to design a Cloudant model.
+
+### Careful partial update
+
+*Cloudant does not support the idea of updating a document. All "updates" on a document are _destructive_ replacements.*
 
-## Key features
+For example:
+
+```
+// original document
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "1",
+  "prop2": "2",
+}
+
+// data to be updated
+ds.updateOrCreate('User', {
+  prop1: 'updated1',
+}, function (err, res) {});
+
+// document after update
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "updated1",
+}
+```
+
+*Please note how property `prop2` was completely dropped upon update.*
+
+Workaround: Do not pass partial values for the data object to be updated. If there are properties that are not being updated, please include the old value to keep data persistent.
 
-* Uses Cloudant Query (Lucene) to support ad-hoc searching.
-* [Loopback query](http://loopback.io/doc/en/lb3/Querying-data.html) support for: fields, limit, order, skip and where filters.
-* Query and filtering is performed on the database for optimal efficiency.
-* Use different DB instances per model definition.
-* Supports basic model discovery.
+We have some discussion on update methods, the issue link can be found in [Feature Backlog](https://github.com/strongloop/loopback-connector-cloudant##feature-backlog) section
 
-## Installation
+### Careful Frequent Modification
+
+Cloudant is not designed to change same document frequently and multiple times. It stores status change by creating different documents and including the same unique id to tell that they are attached to the same item, not updating the same document.
```

Comment:
Change this to: `Cloudant is not designed to change the same document multiple times.`